libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -161,30 +174,264 @@ def get_node_topology(self, node_id=None, auth_header=None):
         response = self.get_system_topology(uri_param, auth_header)
         return response
 
-    #Function Not Ready
-    def verify_nodes(self, node_id=None, expected_response=HTTPStatus.OK):
+    def verify_node_topology(self, expected_response: str):
         """
-        Verify number of nodes and node names
+        Verify node topology
         """
-        resp = self.get_node_topology(node_id)
-        solution_yaml = self.get_solution_yaml()
-        result = resp.status_code == expected_response
+        get_resp = self.get_node_topology()
+        self.log.info("Node topology response %s ", get_resp.json())
+        result = get_resp.status_code == expected_response
         if result:
-            get_response = resp.json()
-            self.log.info("Verify node names")
-            resp_hostnames = []
-            for names in get_response["topology"]["nodes"]:
-                resp_hostnames.append(names["hostname"])
-            #compare for hostnames list received from each pod
-            self.log.info("Verify number of nodes in resp and solution yaml")
-            nodes = len(get_response["topology"]["nodes"])
-            num_nodes = len(solution_yaml["solution"]["storage_set"]["nodes"])
-            if num_nodes != nodes:
-                self.log.error("Actual and expected response for number of nodes didnt match")
+            self.log.info("Step 1: Verify node id and hostname")
+            result, err_msg = self.verify_id_and_hostname()
+            assert result, err_msg
+            self.log.info("Step 2: Verify services list")
+            result, err_msg = self.verify_services_list(get_resp)
+            assert result, err_msg
+        else:
+            err_msg = "Status code check failed"
+            self.log.error(err_msg)
+        return result, err_msg
+
+    def verify_unique_ids(self):
+        """
+        Function to verify cluster ids have unique values
+        """
+        err_msg = ""
+        result = True
+        node_id_list, hostname_list = self.generate_node_hostname_list()
+        self.log.info("List of hostnames: %s ", hostname_list)
+        self.log.info("List of unique ids: %s ", node_id_list)
+        if len(set(node_id_list)) == len(node_id_list):
```

Comment:
both are node_id_list ?

Reply:
yes this is to check if node_id_list has only unique elements

Reply:
ok